.pylintrc

```diff
@@ -67,6 +67,7 @@ enable=all
 # no Warning level messages displayed, use "--disable=all --enable=classes
 # --disable=W".
 disable=W0212,
+  W0231,
```

Comment:
I may be missing something here, because I'm not able to replicate.

I'm running:

```
# python3 -V
Python 3.11.0
# pylint --version
pylint 2.13.7
astroid 2.11.7
Python 3.11.0 (main, Oct 24 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-2)]
```

And `W0231` seems to be enabled indeed:

```
# pylint --list-msgs-enabled  | grep W0231
  super-init-not-called (W0231)
```

But, even without this change:

```
# git rev-parse HEAD
27f18c2cd4465488fbc7f1fbdbab599f57603f7d
```

I'm able to lint the file you mention without issues:

```
# pylint --rcfile=.pylintrc avocado/core/plugin_interfaces.py 

------------------------------------
Your code has been rated at 10.00/10
```

Can you double check things on your side and help me to reproduce it here?

Reply:
Hi @clebergnu, yes, your results are correct, maybe I didn't explain the problem well. The issue is not the `avocado/core/plugin_interfaces.py` itself, but classes who inherit from those interfaces. So if you run the check to the whole avocado files, you will get:
```
** Running lint...
************* Module avocado.plugins.assets
avocado/plugins/assets.py:255:4: W0231: __init__ method from base class 'JobPreTests' is not called (super-init-not-called)
************* Module avocado.plugins.diff
avocado/plugins/diff.py:43:4: W0231: __init__ method from base class 'CLICmd' is not called (super-init-not-called)
************* Module avocado.plugins.journal
avocado/plugins/journal.py:52:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)
************* Module avocado.plugins.sysinfo
avocado/plugins/sysinfo.py:158:4: W0231: __init__ method from base class 'JobPostTests' is not called (super-init-not-called)
************* Module avocado.plugins.testlogs
avocado/plugins/testlogs.py:115:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)
************* Module avocado.plugins.spawners.podman
avocado/plugins/spawners/podman.py:104:4: W0231: __init__ method from base class 'DeploymentSpawner' is not called (super-init-not-called)
************* Module avocado.utils.asset
avocado/utils/asset.py:181:31: E0602: Undefined variable 'self' (undefined-variable)
************* Module resultsdb
optional_plugins/resultsdb/avocado_resultsdb/resultsdb.py:38:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)

------------------------------------------------------------------
Your code has been rated at 9.99/10 (previous run: 9.99/10, +0.00)


--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

************* Module avocado.plugins.bystatus
avocado/plugins/bystatus.py:23:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)
************* Module avocado.plugins.journal
avocado/plugins/journal.py:52:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)
************* Module avocado.plugins.beaker_result
avocado/plugins/beaker_result.py:34:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)
************* Module avocado.plugins.sysinfo
avocado/plugins/sysinfo.py:158:4: W0231: __init__ method from base class 'JobPreTests' is not called (super-init-not-called)
avocado/plugins/sysinfo.py:158:4: W0231: __init__ method from base class 'JobPostTests' is not called (super-init-not-called)
************* Module avocado.plugins.tap
avocado/plugins/tap.py:57:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)
************* Module avocado.plugins.testlogs
avocado/plugins/testlogs.py:115:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)
************* Module avocado.plugins.teststmpdir
avocado/plugins/teststmpdir.py:31:4: W0231: __init__ method from base class 'JobPre' is not called (super-init-not-called)
avocado/plugins/teststmpdir.py:31:4: W0231: __init__ method from base class 'JobPost' is not called (super-init-not-called)
************* Module avocado.plugins.diff
avocado/plugins/diff.py:43:4: W0231: __init__ method from base class 'CLICmd' is not called (super-init-not-called)
************* Module avocado.utils.asset
avocado/utils/asset.py:181:31: E0602: Undefined variable 'self' (undefined-variable)
************* Module avocado.plugins.spawners.podman
avocado/plugins/spawners/podman.py:104:4: W0231: __init__ method from base class 'DeploymentSpawner' is not called (super-init-not-called)
************* Module resultsdb
optional_plugins/resultsdb/avocado_resultsdb/resultsdb.py:38:4: W0231: __init__ method from base class 'ResultEvents' is not called (super-init-not-called)

-------------------------------------------------------------------
Your code has been rated at 9.99/10 (previous run: 10.00/10, -0.00)

```

Reply:
After our discussion, I disable it only for classes which use plugin_interface.